src/avram/define_attribute.cr

```diff
@@ -100,4 +100,57 @@ module Avram::DefineAttribute
       end
     end
   end
+
+  macro file_attribute(key)
+    {% unless key.is_a?(SymbolLiteral) %}
+      {% raise "file_attribute must be declared with a Symbol" %}
+    {% end %}
+
```

Comment:
In the `attribute` macro, there is a line with

```cr
{% ATTRIBUTES << type_declaration %}
```

I believe it's for inheritance? This is another thing I couldn't figure out. You can't create a type declaration in a macro, e.g.:

```cr
ATTRIBUTES << {{name}} : Avram::Uploadable
```

So I'll need some help here. :)

Reply:
Solved!